src/lib.rs

```diff
@@ -200,7 +210,7 @@ impl<'a, R> Iterator for PacketsIter<'a, R> where R: 'a + Read + Seek {
             },
 
             len => {
-                buffer.truncate(len as usize);
+                buffer.truncate(len as usize / 2);
```

Comment:
`len` returned by `ov_read` is in bytes, whereas buffer contains `u16`s, hence the `/2`. Just like `buffer_len` above is `buffer.len() * 2`